psydac/linalg/sparse.py

```diff
@@ -0,0 +1,107 @@
+# coding: utf-8
+
+from scipy.sparse import csr_matrix
+
+from psydac.linalg.basic   import LinearOperator
+from psydac.linalg.basic   import VectorSpace, Vector, LinearOperator
+from psydac.linalg.stencil import StencilVector
+from psydac.linalg.block   import BlockVector
+
+__all__ = (
+    'SparseMatrixLinearOperator',
+)
+
+class SparseMatrixLinearOperator(LinearOperator):
+    """ 
+    LinearOperator representation of a sparse matrix.
+
+    Parameters
+    ----------
+    domain : VectorSpace
+        The domain of the operator.
+
+    codomain : VectorSpace
+        The codomain of the operator.
+
+    sparse_matrix : scipy.sparse.csr_matrix
+        The sparse matrix representing the operator.
```

Comment:
Does it have to be a CSR matrix? If any SciPy sparse matrix is accepted, one can relax the docstring and the type check, and perform the conversion to CSR in the `__init__`. Since what we want to do is a matrix-vector product, the recommended formats in terms of efficiency are CSR, BSR, and DIA. Further, we should keep in mind that SciPy is migrating from sparse matrices to sparse arrays (see [here](https://docs.scipy.org/doc/scipy/reference/sparse.html)). I am suggesting an improvement in a few places.
```suggestion
    sparse_matrix : scipy.sparse.sparray | scipy.sparse.spmatrix
        The sparse SciPy matrix representing the operator. Recommended formats are
        CSR, BSR, and DIA. Any other format will be converted to CSR (csr_array).
```